src/Features/VisualBasic/Portable/ReplaceMethodWithProperty/VisualBasicReplaceMethodWithPropertyService.vb

```diff
@@ -0,0 +1,233 @@
+            End If
+
+            Return containingMethod
+        End Function
+
+        Public Sub RemoveSetMethod(editor As SyntaxEditor, setMethodDeclaration As SyntaxNode) Implements IReplaceMethodWithPropertyService.RemoveSetMethod
+            Dim setMethodStatement = TryCast(setMethodDeclaration, MethodStatementSyntax)
+            If setMethodStatement Is Nothing Then
+                Return
+            End If
+
+            Dim methodOrBlock = GetParentIfBlock(setMethodStatement)
+            editor.RemoveNode(methodOrBlock)
+        End Sub
+
+        Public Sub ReplaceGetMethodWithProperty(
+            editor As SyntaxEditor,
+            semanticModel As SemanticModel,
+            getAndSetMethods As GetAndSetMethods,
+            propertyName As String, nameChanged As Boolean) Implements IReplaceMethodWithPropertyService.ReplaceGetMethodWithProperty
+
+            Dim getMethodDeclaration = TryCast(getAndSetMethods.GetMethodDeclaration, MethodStatementSyntax)
+            If getMethodDeclaration Is Nothing Then
+                Return
+            End If
+
+            Dim methodBlockOrStatement = GetParentIfBlock(getMethodDeclaration)
+            editor.ReplaceNode(methodBlockOrStatement,
+                               ConvertMethodsToProperty(editor, semanticModel, getAndSetMethods, propertyName, nameChanged))
+        End Sub
+
+        Private Function GetParentIfBlock(declaration As MethodStatementSyntax) As DeclarationStatementSyntax
+            If declaration.IsParentKind(SyntaxKind.FunctionBlock) OrElse declaration.IsParentKind(SyntaxKind.SubBlock) Then
+                Return DirectCast(declaration.Parent, DeclarationStatementSyntax)
+            End If
+
+            Return declaration
+        End Function
+
+        Private Function ConvertMethodsToProperty(
+            editor As SyntaxEditor,
+            semanticModel As SemanticModel,
+            getAndSetMethods As GetAndSetMethods,
+            propertyName As String, nameChanged As Boolean) As DeclarationStatementSyntax
+
+            Dim generator = editor.Generator
+
+            Dim getMethodStatement = DirectCast(getAndSetMethods.GetMethodDeclaration, MethodStatementSyntax)
+            Dim setMethodStatement = TryCast(getAndSetMethods.SetMethodDeclaration, MethodStatementSyntax)
+
+            Dim propertyNameToken = GetPropertyName(getMethodStatement.Identifier, propertyName, nameChanged)
+
+            Dim newPropertyDeclaration As DeclarationStatementSyntax
+            If getAndSetMethods.SetMethod Is Nothing Then
+                Dim modifiers = getMethodStatement.Modifiers.Add(SyntaxFactory.Token(SyntaxKind.ReadOnlyKeyword))
+                Dim propertyStatement = SyntaxFactory.PropertyStatement(
+                        getMethodStatement.AttributeLists, modifiers, propertyNameToken, Nothing,
+                        getMethodStatement.AsClause, initializer:=Nothing, implementsClause:=getMethodStatement.ImplementsClause)
```

Comment:
I'm not sure i understand the question.  

Currently the feature only allows replacing a simple method with a property.  This is partly because the feature works across language.  If we allowed you to replace a parameterized method with a parameterized property in VB, then we'd have to uglify your C# code.

I also don't think there's huge value in providing a refactoring from parameterized method to parameterized property.  If you do that, you don't end up needing ot fix up callsites.  Whereas with "normal method" to "normal property" the refactoring saves you a lot of time.  

Note: we filter out parameterized properties very early in the process.  So we don't even get here for those types of properties.